pygfx/resources/_buffer.py

```diff
@@ -6,27 +6,37 @@
 
 
 class Resource(Trackable):
+    """Resource base class."""
+
     pass
 
 
 class Buffer(Resource):
-    """A buffer object represents a piece of memory to the GPU, that can be
-    used as index buffer, vertex buffer, uniform buffer, or storage buffer.
-    You can provide (and update data for it), or use it as a placeholder
-    for a buffer with no representation on the CPU.
-
-    Parameters:
-        data (array, optional): Array data of any type that supports the
-            buffer-protocol, (e.g. a bytes or numpy array). If not given
-            or None, nbytes and nitems must be provided. The data is
-            copied if it's float64 or not contiguous.
-        nbytes (int): The number of bytes. If data is given, it is derived.
-        nitems (int): The number of items. If data is given, it is derived.
-        format (str): The format to use when used as a vertex buffer.
-            By default this is automatically set from the data. This
-            must be a pygfx format specifier, e.g. "3xf4", but can also
-            be a format specific to the render backend if necessary
-            (e.g. from ``wgpu.VertexFormat``).
+    """A contigous piece of GPU memory.
+
+    Buffers can be used as index buffer, vertex buffer, uniform buffer, or
+    storage buffer. You can provide (and update data for it), or use it as a
+    placeholder for a buffer with no representation on the CPU.
+
+    @almarklein: does this mean we differentiate between paged, pinned, and
+    device (GPU) memory?
+
```

Comment:
Okay, so we don't differentiate explicitly in pygfx? I'm asking because this is a very big topic for deep learning and working with CUDA and I would have expected something similar for rendering.

If pygfx doesn't deal with this concept internally (or expose it in the Buffer class) then we can just leave it for now.

Reply:
There is no way to specify these things in wgpu. We can only select whether it's a uniform, vertex, read-only-storage, regular storage. And we can specify where/how the buffer is used.